crypto/crypto/src/ed25519.rs

```diff
@@ -378,7 +397,9 @@ impl TryFrom<&[u8]> for Ed25519PublicKey {
         // Unfortunately, tuple struct `PublicKey` is private so we cannot
         // Ok(Ed25519PublicKey(ed25519_dalek::PublicKey(compressed, point)))
         // and we have to again invoke deserialization.
-        Ed25519PublicKey::from_bytes_unchecked(bytes)
+        let key = Ed25519PublicKey::from_bytes_unchecked(bytes)?;
+        add_tag!(&key, ValidPublicKeyTag); // This key has gone through validity checks.
```

Comment:
Why not using a string here instead of defining a type `ValidPublicKeyTag`?

Reply:
The current tagging mechanism requires explicitly defined tag types. These types can be parameterized by a propagation set, which specifies the set of operations (e.g., bit-xor, subcomponents) that can propagate the tag.

I'll make the tag definition more modular.